code/modules/power/rbmk/rbmk_procs.dm

```diff
@@ -0,0 +1,613 @@
+ */
+/obj/machinery/atmospherics/components/unary/rbmk/core/proc/send_radio_explanation()
+	if(warning_damage_flags & RBMK_PRESSURE_DAMAGE)
+		radio.talk_into(src, "Warning: Reactor overpressurized! Integrity: [get_integrity_percent()]%", engineering_channel)
+		warning_damage_flags &= RBMK_PRESSURE_DAMAGE
+		warning_damage_flags &= RBMK_TEMPERATURE_DAMAGE //If it is both overpressurized and overheating, just send the more important message
+	else if(warning_damage_flags & RBMK_TEMPERATURE_DAMAGE)
+		radio.talk_into(src, "Warning: Reactor overheating! Integrity: [get_integrity_percent()]%", engineering_channel)
+		warning_damage_flags &= RBMK_TEMPERATURE_DAMAGE
+
+/**
+ * Called by check_alert() in this file
+ * Called when the damage has reached critical levels, start the countdown before the destruction, calls meltdown()
+ */
+/obj/machinery/atmospherics/components/unary/rbmk/core/proc/countdown()
+	set waitfor = FALSE
+
+	if(final_countdown) // We're already doing it go away
+		return
+	final_countdown = TRUE
+	var/speaking = "[emergency_alert] The RBMK has reached critical integrity failure. Emergency control rods lowered."
+	radio.talk_into(src, speaking, common_channel, language = get_selected_language())
+
+	notify_ghosts(
+		"The [src] has begun melting down!",
+		source = src,
+		header = "Meltdown Incoming",
+		ghost_sound = 'sound/machines/warning-buzzer.ogg',
+		notify_volume = 75,
+	)
+
+	for(var/i in REACTOR_COUNTDOWN_TIME to 0 step -10)
+		if(critical_threshold_proximity < melting_point) // Cutting it a bit close there engineers
+			radio.talk_into(src, "[safe_alert] Failsafe has been disengaged.", common_channel)
+			final_countdown = FALSE
+			return
+		else if((i % 50) != 0 && i > 50) // A message once every 5 seconds until the final 5 seconds which count down individualy
+			sleep(1 SECONDS)
+			continue
+		else if(i > 50)
+			if(i == 5 SECONDS)
+				sound_to_playing_players('sound/effects/rbmk/explode.ogg')
+			speaking = "[DisplayTimeText(i, TRUE)] remain before total integrity failure."
+		else
+			speaking = "[i*0.1]..."
+		radio.talk_into(src, speaking, common_channel)
+		sleep(1 SECONDS)
+
+	if(pressure > RBMK_PRESSURE_CRITICAL)
+		blowout()
+	else if(temperature > RBMK_TEMPERATURE_CRITICAL)
+		meltdown()
+	else
+		meltdown() //This is caused if neither pressure nor temperature was in critical. We still want to explode
+
+
+/**
+ * Called by countdown() in this file
+ * Create the explosion before deleting the machine core.
+ */
+/obj/machinery/atmospherics/components/unary/rbmk/core/proc/meltdown()
+	set waitfor = FALSE
+	SSair.atmos_machinery -= src //Annd we're now just a useless brick.
+	update_icon()
+	STOP_PROCESSING(SSmachines, src)
+	AddComponent(/datum/component/radioactive, 15000 , src)
+	var/turf/reactor_turf = get_turf(src)
+	var/rbmkzlevel = reactor_turf.get_virtual_z_level()
+	for(var/mob/player_mob in GLOB.player_list)
+		if(player_mob.get_virtual_z_level() == rbmkzlevel)
```

Comment:
```suggestion
		if(compare_z(rbmkzlevel, player_mob.get_virtual_z_level())
```
This is a proper way to compare Zs. getting z value only tells the number, not checking if they're in the same z category. (i.e. station)